Fix e66cec8f86: Permit loading of industry production callback with invalid cargo type.

PeterN opened the pull request:
It is only an error if the invalid result is actually used. This will be silently ignored at the moment.

andythenorth commented:
This now loads the test grf from https://github.com/OpenTTD/nml/files/2949781/firs.grf.zip

PeterN committed:
Fix e66cec8f86: Permit loading of industry production callback with invalid cargo type.

It is only an error if the invalid result is actually used. This will be silently ignored at the moment.
It is still an error if a duplicate cargo type is returned.

andythenorth commented:
Still works after force-push update.

PeterN committed:
Cleanup: Update changed string in language files.

nielsmh commented on src/newgrf_industries.cpp:
```diff
@@ -605,6 +605,16 @@ void IndustryProductionCallback(Industry *ind, int reason)
+			/* display error message */
+			SetDParamStr(0, spec->grf_prop.grffile->filename);
+			SetDParam(1, spec->name);
+			ShowErrorMessage(STR_NEWGRF_BUGGY, STR_NEWGRF_BUGGY_INVALID_CARGO_PRODUCTION_CALLBACK, WL_WARNING);
```
Would it be possible to include the coordinates of the specific industry in the error?
